spacy/pipeline/edit_tree_lemmatizer.py

```diff
@@ -186,8 +199,12 @@ def _scores2guesses(self, docs, scores):
 
         return guesses
 
-    def set_annotations(self, docs: Iterable[Doc], batch_tree_ids):
+    def set_annotations(self, docs: Iterable[Doc], activations: ActivationsT):
+        batch_tree_ids = activations["guesses"]
         for i, doc in enumerate(docs):
+            doc.activations[self.name] = {}
+            for activation in self.store_activations:
+                doc.activations[self.name][activation] = activations[activation][i]
```

Comment:
Fixed.

Reply:
I reverted the change. With the dict comprehension mypy infers that the type is `Union[Floats2d, Ints1d]`, unfortunately it's not able to widen it to the necessary `Union[ArrayXd, Ragged]]` in the assignment to `doc.activations`. I think my original code passes the test because the initial

```
doc.activations[self.name] = {}
```

makes MyPy infer the dictionary type as `Union[ArrayXd, Ragged]]`.